eureka2-core/src/main/java/com/netflix/eureka2/channel/AbstractServiceChannel.java

```diff
@@ -60,4 +66,27 @@ public void onNext(T message) {
             }
         });
     }
+
+    protected boolean moveToState(STATE from, STATE to) {
```

Comment:
Thanks for spotting this. It is a bug. I wanted to provide common implementation that both handles state transitions and related state metrics. I need to remove old implementation.